packages/core/src/views/connect/Index.svelte

```diff
@@ -22,87 +25,103 @@
   import InstallWallet from './InstallWallet.svelte'
   import SelectingWallet from './SelectingWallet.svelte'
   import Sidebar from './Sidebar.svelte'
+  import type {
+    ConnectOptions,
+    i18n,
+    WalletState,
+    WalletWithLoadingIcon
+  } from '../../types'
 
-  export let autoSelect: string
+  export let autoSelect: ConnectOptions['autoSelect']
 
   const { walletModules, appMetadata } = internalState$.getValue()
 
-  let loading = true
   let connectionRejected = false
   let wallets: WalletWithLoadingIcon[] = []
   let selectedWallet: WalletState | null
   let agreed: boolean
+  let connectingWallet: string // the wallet label that is connecting
```

Comment:
you could add `label` to the variable and then the comments wouldnt be needed

Reply:
Yeah good call